darts/utils/timeseries_generation.py

```diff
@@ -206,10 +206,58 @@ def random_walk_timeseries(length: int = 10,
     return TimeSeries.from_times_and_values(times, values, freq=freq)
 
 
+def _extend_time_index_until(time_index: Union[pd.DatetimeIndex, pd.RangeIndex],
+                             until: Optional[Union[int, str, pd.Timestamp]],
+                             add_length: int,
+                             ) -> pd.DatetimeIndex:
+
+    if not add_length and not until:
+        return time_index
+
+    raise_if(add_length and until, "set only one of add_length and until")
+
+    end = time_index[-1]
+    freq = time_index.freq
+
+    if add_length > 0:
+        raise_if_not(add_length >= 0, "until, indicating the number of time points to extend into the future by, must be positive")
```

Comment:
I guess we don't need this one, since no error will ever be raised due to the if condition before

Reply:
You could instead add a check before the if condition to make sure no negative value was provided for `add_length`